packages/fuels-contract/src/script_calls.rs

```diff
@@ -0,0 +1,181 @@
+use std::fmt::Debug;
+use std::marker::PhantomData;
+
+use fuel_gql_client::fuel_tx::{Output, Receipt, Transaction};
+use fuel_tx::Input;
+
+use crate::contract::get_decoded_output;
+
+use fuels_core::{
+    parameters::{CallParameters, TxParameters},
+    Tokenizable,
+};
+use fuels_signers::{provider::Provider, WalletUnlocked};
+
+use fuels_types::{errors::Error, param_types::ParamType};
+
+use crate::execution_script::{CompiledScript, TransactionExecution};
+
+pub struct ScriptInterface {
+    pub script: TransactionExecution,
+    pub wallet: WalletUnlocked,
+}
+
+#[derive(Debug)]
+pub struct ScriptCallResponse<D> {
+    pub value: D,
+    pub receipts: Vec<Receipt>,
+    pub gas_used: u64,
+}
+
+impl<D> ScriptCallResponse<D> {
+    /// Get the gas used from ScriptResult receipt
+    fn get_gas_used(receipts: &[Receipt]) -> u64 {
+        receipts
+            .iter()
+            .rfind(|r| matches!(r, Receipt::ScriptResult { .. }))
+            .expect("could not retrieve ScriptResult")
+            .gas_used()
+            .expect("could not retrieve gas used from ScriptResult")
+    }
+
+    pub fn new(value: D, receipts: Vec<Receipt>) -> Self {
+        Self {
+            value,
+            gas_used: Self::get_gas_used(&receipts),
+            receipts,
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Contains all data relevant to a single script call
+pub struct ScriptCall {
+    pub compiled_script: CompiledScript,
+    pub script_data: Vec<u8>,
+    pub inputs: Vec<Input>,
+    pub outputs: Vec<Output>,
+    pub call_parameters: CallParameters,
```

Comment:
I left it there because I misunderstood what `CallParameters` are for and used in. Do you confirm that they would not be useful in a script call, even if the script calls some contracts?

Reply:
In the case of contract calls, `CallParameters` are possible because we control what addresses the script data is loaded from since we build the calling script fusing fuel asm opcodes.
When we load a script directly from a binary, we have no control over the addresses the data is being loaded from.

I am sure of the above, but I wonder what would happen if we were to load a binary of a script that performs a call. Does forc embed the data into the binary?

Reply:
This is outside my knowledge, @digorithm  @mohammadfawaz?

Reply:
I'm also unsure here, will do some digging.

> I am sure of the above, but I wonder what would happen if we were to load the binary of a script that performs a call. Does forc embed the data into the binary?

@iqdecay can you test this?

Reply:
I will test this in a PR related to contract calls. Leaving the `call_parameters` field here for now.